### PR TITLE
Update go version to 1.22.7 to fix [CVE-2024-34156],[CVE-2024-34155]

### PR DESCRIPTION
**Pull request by Rchanger**
update go version

**What this PR does / why we need it**:
https://github.com/helm/helm/issues/13318

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains user facing changes (the `docs needed` label should be applied if so)
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Comment by kanchwala-yusuf** [comment]
@mattfarina, @robertsirc Can you please approve this PR? This PR fixes some of the CVEs https://nvd.nist.gov/vuln/detail/CVE-2024-34156

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module helm.sh/helm/v3
 
-go 1.22.0
+go 1.22.7
```
Comment (mattfarina) [inline go.mod:3]: This changes the _minimum_ version used to build Helm from a compatibility standpoint. That's what the `go` directive.

We don't want to change the minimum version of compatibility. We want to build with the newer version.

**Review by mattfarina — CHANGES_REQUESTED** [review]
The build system should use the latest patch version of Go when we do a release. A release was created today and _should_ have used Go 1.22.7. For some reason, that did not happen. We are investigating the issue.

Updates through the build system is what we use instead of putting specific patch versions as the minimum version in the go.mod file.

File: go.mod (hunk 1 above)
Comment (Rchanger) [reply go.mod:3]: got it will close this PR, thanks @mattfarina

**Pull request closed without merging**